shell/browser/api/electron_api_global_shortcut.cc

```diff
@@ -103,13 +115,59 @@ bool GlobalShortcut::Register(const ui::Accelerator& accelerator,
   }
 #endif
 
-  if (!GlobalShortcutListener::GetInstance()->RegisterAccelerator(accelerator,
-                                                                  this)) {
+  auto* instance = GlobalShortcutListener::GetInstance();
+
+  if (!instance) {
+    LOG(ERROR) << "GlobalShortcuts are not available";
     return false;
   }
 
-  accelerator_callback_map_[accelerator] = callback;
-  return true;
+  if (instance->IsRegistrationHandledExternally()) {
+    auto* context = ElectronBrowserContext::From("", false);
+    PrefService* prefs = context->prefs();
+
+    // Need a unique profile id. Set one if not generated yet, otherwise re-use
+    // the same so that the session for the globalShortcuts is able to get
+    // already registered shortcuts from the previous session. This will be used
+    // by GlobalShortcutListenerLinux as a session key.
+    std::string profile_id = prefs->GetString(kElectronGlobalShortcutsUuid);
+    if (profile_id.empty()) {
+      auto uuid = base::Uuid::GenerateRandomV4();
+      profile_id = uuid.AsLowercaseString();
+      prefs->SetString(kElectronGlobalShortcutsUuid, profile_id);
+    }
```

Comment:
nit: we don't use `uuid` again so this can be 

```suggestion
    std::string profile_id = prefs->GetString(kElectronGlobalShortcutsUuid);
    if (profile_id.empty()) {
      profile_id = base::Uuid::GenerateRandomV4().AsLowercaseString();
      prefs->SetString(kElectronGlobalShortcutsUuid, profile_id);
    }
```

Reply:
Done.